docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc

```diff
@@ -1,12 +1,115 @@
+clocks and calendars. We've invented leap years and leap seconds, standard and
+daylight savings times, and timezone offsets of 30 or 45 minutes rather than a 
+full hour. While these creations help keep us in sync with the cosmos and our
+environment, they can make specifying time intervals accurately a real challenge.
+The only universal truth our researchers have yet to disprove is that a
+millisecond is always the same duration, and a second is always 1000 milliseconds.
+Beyond that, things get complicated.
+
+Generally speaking, when you specify a single time unit, such as 1 hour or 1 day, you
+are working with a _calendar interval_, but multiples, such as 6 hours or 3 days, are
+_fixed-duration intervals_.
+
+For example, a specification of 1 day (1d) from now is a calendar interval that
+means "at
+this exact time tomorrow" no matter the length of the day. A change to or from 
+daylight savings time that results in a 23 or 25 hour day is compensated for and the
+specification of "this exact time tomorrow" is maintained. But if you specify 2 or
+more days, each day must be of the same fixed duration (24 hours). In this case, if
+the specified interval includes the change to or from daylight savings time, the
+interval will end an hour sooner or later than you expect.
+
+There are similar differences to consider when you specify single versus multiple
+minutes or hours. Multiple time periods longer than a day are not supported.
+
+Here are the valid time specifications and their meanings:
+
+*milliseconds (ms)* +
+Fixed length interval; supports multiples.
+
+*seconds (s)* +
+1000 milliseconds; fixed length interval (except for the last second of a
+minute that contains a leap-second, which is 2000ms long); supports multiples.
+
+*minutes (m)* +
+All minutes begin at 00 seconds. +
+* One minute (1m) is the interval between 00 seconds of the first minute and 00
+seconds of the following minute in the specified timezone, compensating for any
+intervening leap seconds, so that the number of minutes and seconds past the 
+hour is the same at the start and end.
+* Multiple minutes (_n_m) are intervals of exactly 60x1000=60,000 milliseconds
+each.
+
+*hours (h)* +
+All hours begin at 00 minutes and 00 seconds. +
+* One hour (1h) is the interval between 00:00 minutes of the first hour and 00:00
+minutes of the following hour in the specified timezone, compensating for any
+intervening leap seconds, so that the number of minutes and seconds past the hour is the same at the start and end.
+* Multiple hours (_n_h) are intervals of exactly 60x60x1000=3,600,000 milliseconds each.
+
+*days (d)* +
+All days begin at the earliest possible time, which is usually 00:00:00
+(midnight).
+* One day (1d) is the interval between 00:00:00 of the first day and 00:00:00
+of the following day in the specified timezone, compensating for any intervening
+time changes, so that the number of hours, minutes, and seconds of the day is
```

Comment:
I'm not sure where you got "consecutive days containing timezone transitions", and Colin approved this. If I change "time changes" to "time change", will that sound better to you?

Reply:
No, the issue is with days that contain no midnight. For instance, São Paolo set their clocks forwards from `-3:00` to `-2:00` at `0300 UTC` on `2017-10-15`, so in local time `2017-10-14 23:59:59` is followed by `2017-10-15 01:00:00`. In this case the nearby buckets that we generate are:

- `2017-10-14T00:00:00.000-03:00` (24 hours long)
- `2017-10-15T01:00:00.000-02:00` (23 hours long)
- `2017-10-16T00:00:00.000-02:00` (24 hours long)

Thus the day is not always the interval between midnight of the first day and midnight of the second, and the hours of the start and end time of the bucket are not always equal. However, at least one of the start or the end is at midnight, because there's never two consecutive days without a midnight.

Reply:
I re-read my original comment and it does seems rather unclear what I actually meant. Sorry about that - hopefully this helps.